### PR TITLE
Add option to disable archive with Gym

### PR DESCRIPTION
**Pull request by dylangyesbreghs**
Disable archive while using gym.
As a company, we use this to build or development pods.

### CODE REVIEW

**Comment by KrauseFx** [comment]
`gym`'s goal is to generate an ipa file, so that doesn't really fit into its mission. Can you explain closer on why you'd like this change?

**Comment by dylangyesbreghs** [comment]
We currently have a complex application were al the different component are tested. But we want to make sure the entire application still build with the integrated components. Therefore we just want to build the application, and not archive it.

Archiving often gives use code signing issues. Unfortunately `match` can't be used because we are on a enterprise account.

We prefer not to use `xcodebuild` because we like to use `gym`.

**Comment by dreyhomedev** [comment]
In my understanding you need an archive to build an ipa.
If your goal is to avoid storing the archive in xcode organizer simply alter the archive path in gym.

But maybe i'm missing something?

**Comment by fousa** [comment]
We could also use `xcodebuild`. This allows us to just build the application without generating an IPA or archive. But then we get the deprecation warning.

Is `gym` a replacement action for `xcodebuild`?

File: lib/gym/generators/build_command_generator.rb [hunk 1]
```diff
@@ -90,16 +92,19 @@ def build_path
...
+            Gym.cache[:archive_path] = File.join(build_path, file_name.join(" ") + ".xcarchive")
+          end
+
+          if File.extname(Gym.cache[:archive_path]) != ".xcarchive"
+            Gym.cache[:archive_path] += ".xcarchive"
+          end
+          return Gym.cache[:archive_path]
         end
-        return Gym.cache[:archive_path]
+        return ""
```
Comment (lacostej) [inline lib/gym/generators/build_command_generator.rb]: Not sure we want this feature, but if we want it, shouldn t it be more logical to add
```
raise "No archive path available when " unless Gym.config[:archive] 
```

at the top of this method ?

**Comment by KrauseFx** [comment]
Hey @dylangyesbreghs, thanks for your contribution, however as briefly mentioned before `gym` was designed to build IPA files, and is not a generic `xcodebuild` runner. You can use the built-in `xcodebuild` action in `fastlane` to build your development pods. I'll close this issue for now, thanks again for your PR :+1:

**Pull request closed without merging**